spec/services/step_factory_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe StepFactory do
+  subject { described_class }
+
+  describe "#create" do
+    context "with a valid constant" do
+      it "instantiates the class" do
+        expect(subject.create("Identity")).to be_an_instance_of(Identity)
+      end
+    end
+
+    context "with an invalid constant" do
+      it "raises a NameError" do
```

Comment:
Might be worth throwing a custom `StepNotFoundError` here, along the lines of:

`Unable to instantiate step: <step_name>`

Reply:
Need to do a bit more work here to register permitted classes, as it stand a user could create a String etc.

Reply:
Have set up a custom error with a redirect to root for now